moveit_kinematics/ikfast_kinematics_plugin/templates/ikfast61_moveit_plugin_template.cpp

```diff
@@ -1393,6 +1555,27 @@ bool IKFastKinematicsPlugin::sampleRedundantJoint(kinematics::DiscretizationMeth
   return true;
 }
 
+void IKFastKinematicsPlugin::transformToChainFrame(Eigen::Affine3d& ik_pose) const
+{
+  if (transform_tip_)
+    ik_pose = ik_pose * group_tip_to_chain_tip_;
+
+  if (transform_base_)
+    ik_pose = chain_base_to_group_base_ * ik_pose;
+}
+
+void IKFastKinematicsPlugin::transformToChainFrame(const geometry_msgs::Pose& ik_pose, KDL::Frame& ik_pose_chain) const
+{
+  Eigen::Affine3d ik_eigen_pose;
+  tf::poseMsgToEigen(ik_pose, ik_eigen_pose);
+
+  transformToChainFrame(ik_eigen_pose);
+
+  geometry_msgs::Pose ik_msg_pose;
+  tf::poseEigenToMsg(ik_eigen_pose, ik_msg_pose);
+  tf::poseMsgToKDL(ik_msg_pose, ik_pose_chain);
```

Comment:
add comments for these steps, future you will thank you. (so will others)

Reply:
done